### ISSUE TITLE
VoiceOver does not reads out any Soft keyboard letter after double tapping it on iOS during editing a TextFormField

### ISSUE THREAD

**Issue by aalieco** [flutter/flutter#61005]
## Steps to Reproduce

```
TextFormField(
            decoration: InputDecoration(
                hintText: Util.isScreenReader(context) ? null : 'Login name',
                labelText: 'Login name',
            ),
            validator: (value) {
              if (value.isEmpty) {
                return 'Please enter login name';
              }
              return null;
            },
            controller: usernameController
          ),
```

1. Create a TextFormField
1. Run the app
2. Turn on VoiceOver on your iPhone 
3. Navigate to that TextFormField
4. Double tap to insert/edit text
5. When the soft keyboard appears, tap a letter button, ex: A, wait till the reader reads it and reads its hint ex: "Alpha"
6. Then double tap to write that letter in the TextFormField

**Expected results:** 
You would expect that it reads the letter "A" after double tapping it as a feedback
**Actual results:** 
But what happened is there were no sound at all from the reader..
Note that our app's previous version built on 23.4.2020 didn't have this missing voice feedback.
<details>

```
[✓] Flutter (Channel master, 1.20.0-3.0.pre.148, on Mac OS X 10.15.5 19F101, locale en-FI)
    • Flutter version 1.20.0-3.0.pre.148 at /Users/msx/dev/flutter
    • Framework revision 2ab5099436 (37 minutes ago), 2020-07-07 10:59:12 -0700
    • Engine revision d0d6a4c236
    • Dart version 2.9.0 (build 2.9.0-20.0.dev 69aba23371)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/aali/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (2 available)
    • iPhone (mobile) • 1fe4a6053b3d1282ec1f109179391d82e54220 • ios • iOS 12.4.6
    • iPhone 11 (mobile)      • BCF25661-F92F-4CE0-A31A-A14151A2C9     • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)

• No issues found!
````

</details>

**Comment by TahaTesser** [comment]
I can reproduce this issue on XR

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.20.0-8.0.pre.40, on Mac OS X 10.15.5 19F101,
    locale en-GB)
    • Flutter version 1.20.0-8.0.pre.40 at /Users/taha/Code/flutter_master
    • Framework revision 6eaaf1650e (11 hours ago), 2020-07-09 18:04:37 -0700
    • Engine revision 9b3e3410f0
    • Dart version 2.9.0 (build 2.9.0-20.0.dev 06cb010247)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (5 available)
    • SM M305F (mobile)      • 32003c30dc19668f          • android-arm64  •
      Android 10 (API 29)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      13.5.1
    • macOS (desktop)        • macos                     • darwin-x64     • Mac
      OS X 10.15.5 19F101
    • Web Server (web)       • web-server                • web-javascript •
      Flutter Tools
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 83.0.4103.116

• No issues found!

``` 

</details>

**Comment by aalieco** [comment]
Any updates on this issue?

**Comment by yama3nomori** [comment]
I have the same problem.
Does it fixed?

**Comment by huycozy** [comment]
Hi @aalieco, this issue seems to have been fixed on the latest master channel as I cannot reproduce it there. Tested on `iPhone 7, iOS 15.7.2`. Please see the demo below.

<details>
<summary>Demo</summary>


https://user-images.githubusercontent.com/104349824/211789439-0b24143e-54df-4f7f-89eb-150f82a3da62.MP4


</details>

<details>
<summary>flutter doctor -v (master)</summary>

```bash
[!] Flutter (Channel master, 3.7.0-17.0.pre.20, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.0-17.0.pre.20 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b13f83a2ba (57 minutes ago), 2023-01-10 17:41:54 -0800
    • Engine revision ae9e181e30
    • Dart version 3.0.0 (build 3.0.0-108.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 108.0.5359.124

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>

I'll be closing this issue since it has already been fixed on the latest master channel. If you still experience this issue, kindly leave a comment below and I will reopen this. Thank you!
